packages/core/src/components/tabs2/tabs.tsx

```diff
@@ -0,0 +1,285 @@
+/*
+ * Copyright 2015 Palantir Technologies, Inc. All rights reserved.
+ * Licensed under the BSD-3 License as modified (the “License”); you may obtain a copy
+ * of the license at https://github.com/palantir/blueprint/blob/master/LICENSE
+ * and https://github.com/palantir/blueprint/blob/master/PATENTS
+ */
+
+import * as classNames from "classnames";
+import * as PureRender from "pure-render-decorator";
+import * as React from "react";
+// import { findDOMNode } from "react-dom";
+
+import { AbstractComponent } from "../../common/abstractComponent";
+import * as Classes from "../../common/classes";
+import * as Keys from "../../common/keys";
+import { IProps } from "../../common/props";
+import { safeInvoke } from "../../common/utils";
+
+import { ITab2Props, Tab2, TabId } from "./tab";
+import { generateTabPanelId, generateTabTitleId, TabTitle } from "./tabTitle";
+
+// <Tabs id="tabs">
+//     <Tab id="a" title="Alpha">
+//         <h1>first panel</h1>
+//     </Tab>
+//     <Tab id="b" title="Beta">
+//         <h1>second panel</h1>
+//     </Tab>
+//     <input type="text" placeholder="Search..." />
+// </Tabs>
+
+// TODO
+// vertical key bindings? up/dn
+
+type TabElement = React.ReactElement<ITab2Props & { children: React.ReactNode }>;
+
+const TAB_SELECTOR = `.${Classes.TAB}`;
```

Comment:
i guess it depends if you expect it to be useful outside the component. for instance, see `static Tab` [below](https://github.com/palantir/blueprint/pull/722/files#diff-ce03cc94ae3b7ac25aa6ea2639bf0448R93) which is only useful outside the component.

this CSS selector should not be necessary outside as querying the DOM is rarely necessary in React.